### PR TITLE
Closes #2142: Upgrade a-c version to 51

### LINKED ISSUE #2142: Upgrade ac to 51
And gecko nightly version

To verify
`2019-04-24 15:05:14.301 10731-10870/org.mozilla.tv.firefox.gecko.debug E/GeckoProcessManager: Cannot connect to process tab`
Fix available in the latest GV 68 (https://hg.mozilla.org/mozilla-central/rev/a1c8daed8d7f)

### PR DESCRIPTION
**Pull request by psymoon**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
- [x] This PR includes a **CHANGELOG entry** or does not need one
- [x] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

**Comment by mcomella** [comment]
THere's a weird TC error:

> Execution failed for task ':app:transformResourcesWithMergeJavaResForSystemDebug'.
> More than one file was found with OS independent path 'META-INF/atomicfu.kotlin_module'

Not sure what's up with that – I wonder if we can reproduce locally?

File: app/proguard-rules.pro [hunk 1]
```diff
@@ -81,3 +81,5 @@
 # A reference class used by androidx.concurrent.futures.AbstractFuture$UnsafeAtomicHelper
 # Currently consumed (in depenendecies) by mozilla.components.service.freboard.scheduler
 -dontwarn sun.misc.Unsafe
+# To be addressed in https://github.com/mozilla-mobile/android-components/issues/2900
```
Comment (mcomella) [inline app/proguard-rules.pro]: nit: this is in the "AndroidX" category. Maybe we should move it up towards the top, in no section.

Comment (mcomella) [inline app/proguard-rules.pro]: nit: same, link to our internal issue rather than the a-c issue.

File: app/build.gradle [hunk 2]
```diff
@@ -93,6 +93,8 @@ android {
     // See https://developer.android.com/jetpack/docs/release-notes#september_19_2018
     packagingOptions {
         exclude 'META-INF/proguard/androidx-annotations.pro'
+        // Remove when https://github.com/mozilla-mobile/android-components/issues/2900 is addressed
```
Comment (mcomella) [inline app/build.gradle]: nit: can we file our own follow-up issue that tracks the a-c issue and reference that instead? It makes it harder to forget to fix.

Comment (psymoon) [reply app/build.gradle]: Sounds good

**Pull request merged**